### PR TITLE
WEBUI-986: add functional test to capture history data of document

### PR DESCRIPTION
**Pull request by Pranit-Sotre**

### CODE REVIEW

File: ftest/features/history.feature [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    And I login as "John"
+    And I have a Workspace document
+    And I have a File document
+    And I browse to the document
+
+  Scenario: Validate entries in history tab
+    When I can navigate to History pill
+    Then I can see the history table
+    And I have a non empty history table
+    And I can see "Document created" entry in history table
```
Comment (Nisha-QA) [inline ftest/features/history.feature:14]: Blank line missing in the end of the file

Comment (Nisha-QA) [inline ftest/features/history.feature:14]: Validate Download action as well

Comment (Pranit-Sotre) [reply ftest/features/history.feature:14]: Here download action does not takes place, therefore no need to validate.

**Review by Nisha-QA — DISMISSED** [review]
Looks good

**Review by rakeshkumar1019 — DISMISSED** [review]

**Review by GuruprasadAnk1996 — DISMISSED** [review]

**Review by Nisha-QA — DISMISSED** [review]
Looks good

**Review by GuruprasadAnk1996 — DISMISSED** [review]

**Review by Nisha-QA — DISMISSED** [review]
Looks good

**Review by GuruprasadAnk1996 — DISMISSED** [review]

**Review by rakeshkumar1019 — DISMISSED** [review]

**Review by satishyadav-dev — DISMISSED** [review]

File: packages/nuxeo-web-ui-ftest/features/step_definitions/history.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Then } from '@cucumber/cucumber';
+
+Then('I can see {string} entry in history table', function(performedAction) {
+  this.ui.historyTable.waitForVisible();
```
Comment (rajatgoel-gl) [inline packages/nuxeo-web-ui-ftest/features/step_definitions/history.js]: Donot use waitForVisible. Kindly use waitForDisplayed for waitForExist

File: packages/nuxeo-web-ui-ftest/pages/ui/history_table.js [hunk 3]
```diff
@@ -2,6 +2,30 @@ import BasePage from '../base';
...
-    return this.el.element(`///*[text()="${event}"]`);
+    return this.el.$(`///*[text()="${event}"]`);
+  }
+
+  get isHistoryTableVisible() {
+    return this.el.$('#table').waitForVisible();
+  }
+
+  get isHistoryTableFilled() {
+    this.el.waitForVisible('#table nuxeo-data-table-row');
```
Comment (rajatgoel-gl) [inline packages/nuxeo-web-ui-ftest/pages/ui/history_table.js]: Donot use waitForVisible. Kindly use waitForDisplayed for waitForExist

File: packages/nuxeo-web-ui-ftest/pages/ui/history_table.js [hunk 4]
```diff
@@ -2,6 +2,30 @@ import BasePage from '../base';
 
 export default class HistoryTable extends BasePage {
   getHistory(event) {
-    return this.el.element(`///*[text()="${event}"]`);
+    return this.el.$(`///*[text()="${event}"]`);
+  }
+
+  get isHistoryTableVisible() {
+    return this.el.$('#table').waitForVisible();
```
Comment (rajatgoel-gl) [inline packages/nuxeo-web-ui-ftest/pages/ui/history_table.js]: Donot use waitForVisible. Kindly use waitForDisplayed for waitForExist

File: packages/nuxeo-web-ui-ftest/features/step_definitions/history.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  this.ui.historyTable.waitForHasEntry(performedAction).should.be.true;
+});
+
+Then('I can see the history table', function() {
+  this.ui.historyTable.waitForVisible();
+  this.ui.historyTable.isHistoryTableVisible.should.be.true;
```
Comment (rajatgoel-gl) [inline packages/nuxeo-web-ui-ftest/features/step_definitions/history.js:4]: Do we really need multiple visible in this functional test case? Kindly re-check and donot use waitForVisible. Kindly use waitForDisplayed for waitForExist

File: packages/nuxeo-web-ui-ftest/features/step_definitions/history.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+});
+
+Then('I have a non empty history table', function() {
+  this.ui.historyTable.waitForVisible();
+  this.ui.historyTable.isHistoryTableFilled.should.be.true;
```
Comment (rajatgoel-gl) [inline packages/nuxeo-web-ui-ftest/features/step_definitions/history.js:8]: Do we really need multiple visible in this functional test case? Kindly re-check and donot use waitForVisible. Kindly use waitForDisplayed for waitForExist

**Review by rajatgoel-gl — CHANGES_REQUESTED** [review]

**Review by Nisha-QA — APPROVED** [review]

**Review by satishyadav-dev — APPROVED** [review]

**Review by rajatgoel-gl — APPROVED** [review]

**Review by Nishant0928 — APPROVED** [review]

**Review by GuruprasadAnk1996 — APPROVED** [review]

**Review by swarnadipa-dev — APPROVED** [review]

**Pull request merged**